src/core/qgsstacktrace.h

```diff
@@ -0,0 +1,82 @@
+/***************************************************************************
+  qgsstacktrace.h - QgsStackTrace
+
+ ---------------------
+ begin                : 24.4.2017
+ copyright            : (C) 2017 by Nathan Woodrow
+ email                : woodrow.nathan@gmail.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSSTACKTRACE_H
+#define QGSSTACKTRACE_H
+
+#include "qgis_core.h"
+
+#include <QStringList>
+
+/**
+* \ingroup core
+* The QgsStacktrace class provides an interface to generate a stack trace for
+* displaying additional debug information when things go wrong.
+*
+* \note Not available in python
```

Comment:
I'd say this shouldn't be part of stable API at all - if you surround this class with

    \\\@cond PRIVATE
    ...
    \\\@endcond

and do the same in the .cpp file then it gets hidden from doxygen and is no longer fixed api...